src/renderer/vsync/vsync_macos.rs

```diff
@@ -1,60 +1,63 @@
-use std::sync::{Arc, Condvar, Mutex};
-
 use log::{error, trace, warn};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
 
-use crate::renderer::WindowedContext;
+use winit::event_loop::EventLoopProxy;
+
+use crate::{renderer::WindowedContext, window::UserEvent};
 
 use super::macos_display_link::{
     core_video, get_display_id_of_window, MacosDisplayLink, MacosDisplayLinkCallbackArgs,
 };
 
 struct VSyncMacosDisplayLinkUserData {
-    vsync_count: Arc<(Mutex<usize>, Condvar)>,
+    proxy: EventLoopProxy<UserEvent>,
+    redraw_requested: Arc<AtomicBool>,
```

Comment:
Is the `Arc` needed? 

- `Arc` also just uses an `AtomicUsize` under the hood to hold referrers iirc
- `AtomicUsize::swap` and `AtomicUsize::store` take `&self` immutably

so anything shared to could just hold references.

Reply:
No, I think it was a copy paste issue, because I copied the code from the windows implementation. We should be able to remove it.

Reply:
Turns out we can't without doing self-referential magic since `VSyncMacosDisplayLinkUserData` is referenced by `VSyncMacos` itself. I'll drop this then.